src/FactoryTrait.php

```diff
@@ -226,14 +226,14 @@ public function normalizeClassName(string $name, string $prefix = null): string
 
         // Rule 1: if starts with "." always prefix
         if ($name && $name[0] === '.' && $prefix) {
-            $name = $prefix . '\\' . substr($name, 1);
+            $name = $prefix . '\\' . mb_substr($name, 1);
             $name = str_replace('/', '\\', $name);
 
             return $name;
         }
 
         // Rule 2: if "\" is present, don't prefix
-        if (strpos($name, '\\') !== false) {
+        if (mb_strpos($name, '\\') !== false) {
```

Comment:
is is totally not needed, only slower...

Reply:
oh yes you're right